webapp/src/components/App/Singlestudy/explore/Modelization/TableMode/utils.ts

```diff
@@ -0,0 +1,132 @@
+}
+
+export const TABLE_TEMPLATE_TYPE_OPTIONS = Object.values(TableTemplateType);
+
+const TABLE_TEMPLATE_COLUMNS_BY_TYPE = {
+  [TableTemplateType.Area]: [
+    // Optimization - Nodal optimization
+    "nonDispatchablePower",
+    "dispatchableHydroPower",
+    "otherDispatchablePower",
+    "spreadUnsuppliedEnergyCost",
+    "spreadSpilledEnergyCost",
+    // Optimization - Filtering
+    "filterSynthesis",
+    "filterYearByYear",
+    // Adequacy patch
+    "adequacyPatchMode",
+  ],
+  [TableTemplateType.Link]: [
+    "hurdlesCost",
+    "loopFlow",
+    "usePhaseShifter",
+    "transmissionCapacities",
+    "assetType",
+    "linkStyle",
+    "linkWidth",
+    "displayComments",
+    "filterSynthesis",
+    "filterYearByYear",
+  ],
+  [TableTemplateType.Clusters]: [],
+  [TableTemplateType.BindingConstraint]: [],
+} as const;
+
+type TableTemplateColumnsForType<T extends TableTemplateType> = Array<
+  typeof TABLE_TEMPLATE_COLUMNS_BY_TYPE[T][number]
+>;
+
+export interface TableTemplate<
+  T extends TableTemplateType = TableTemplateType
+> {
+  name: string;
+  type: T;
+  columns: TableTemplateColumnsForType<T>;
+}
+
+export function createTableTemplate<T extends TableTemplateType>(
+  name: string,
+  type: T,
+  columns: TableTemplateColumnsForType<T>,
+  toLocalStorage = true
+): TableTemplate<T> {
+  const template: TableTemplate<T> = { name, type, columns };
+
+  if (toLocalStorage) {
+    storage.setItem(StorageKey.StudiesModelTableTemplates, (prev) => [
+      ...(prev || []),
+      template,
+    ]);
+  }
+
+  return template;
+}
+
+export const DEFAULT_TABLE_TEMPLATES: TableTemplate[] = [
+  createTableTemplate(
+    "Economic Opt.",
+    TableTemplateType.Area,
+    [
+      "spreadUnsuppliedEnergyCost",
+      "spreadSpilledEnergyCost",
+      "nonDispatchablePower",
+      "dispatchableHydroPower",
+      "otherDispatchablePower",
+    ],
+    false
+  ),
+];
+
+export function getTableColumnsForType(type: TableTemplateType): string[] {
+  // Arrays have a numeric index signature because of `as const`
+  return Object.values(TABLE_TEMPLATE_COLUMNS_BY_TYPE[type]);
+}
+
+function makeRequestURL(studyId: StudyMetadata["id"]): string {
+  return `v1/studies/${studyId}/config/table_mode`;
+}
+
+export type TableData = Record<
+  string,
+  Record<string, string | boolean | number>
+>;
+
+export async function getTableData<T extends TableTemplateType>(
+  studyId: StudyMetadata["id"],
+  type: T,
+  columns: TableTemplateColumnsForType<T>
+): Promise<TableData> {
+  const res = await client.get(makeRequestURL(studyId), {
+    params: {
+      table_type: type,
+      columns: columns.map(snakeCase).join(","),
+    },
+  });
+  return res.data;
+}
+
+export function setTableData(
+  studyId: StudyMetadata["id"],
+  type: TableTemplateType,
+  data: DeepPartial<TableData>
+): Promise<void> {
+  return client.put(makeRequestURL(studyId), null, {
+    params: {
+      table_type: type,
+      data,
+    },
+  });
+}
```

Comment:
put this in services/api/tablemode.ts

Reply:
I will update all forms in a other ticket